<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to the ASSIGN Package • ASSIGN</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">ASSIGN</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/ASSIGN.vignette.html">Primer</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/compbiomed/ASSIGN">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Introduction to the ASSIGN Package</h1>
                        <h4 class="author">Ying Shen</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA<br><h4 class="author">David Jenkins</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA<br><h4 class="author">W. Evan Johnson</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:wej@bu.edu">wej@bu.edu</a>
      </address>
                  
          </address>
</address>
</div>

    
    
<div class="contents">
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>This vignette provides an overview of the Bioconductor package <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> (Adaptive Signature Selection and InteGratioN) for signature-based profiling of heterogeneous biological pathways. <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> is a computational tool used to evaluate the pathway deregulation/activation status in individual patient samples. <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> employs a flexible Bayesian factor analysis approach that adapts predetermined pathway signatures derived either from a literature search or from perturbation experiments to create cell-/tissue-specific pathway signatures. The deregulation/activation level of each context-specific pathway is quantified to a score, which represents the extent to which a patient sample matches thepathway deregulation/activation signature.</p>
<p>Some distinctive features of <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> are:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Multiple Pathway Profiling</strong>: <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> can profile multiple pathway signatures simultaneously, accounting for ‘cross-talk’ between interconnected pathway components.</li>
<li>
<strong>Context specificity in baseline gene expression</strong>: Baseline gene expression levels (i.e., the gene expression level under normal conditions) may vary widely due to differences across tissue types, disease statuses, or measurement platforms. <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> can adaptively estimate background gene expression levels across a set of samples.</li>
<li>
<strong>Context-specific signature estimation</strong>: <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> provides the flexibility to use either an input gene list or magnitudes of signature genes as prior information, allowing for adaptive refinement of pathway signatures in specific cell or tissue types.</li>
<li>
<strong>Regularization of signature strength estimates</strong>: <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> regularizes the signature strength coefficients using a Bayesian ridge regression formulation by shrinking the strength of the irrelevant signature genes toward zero. The parameter regularization constrains the pathway signature to a small group of genes, making the results more biologically interpretable.</li>
</ol>
</div>
<div id="how-to-use-the-assign-package" class="section level1">
<h1 class="hasAnchor">
<a href="#how-to-use-the-assign-package" class="anchor"></a>How to use the ASSIGN package</h1>
<div id="example-data" class="section level2">
<h2 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example Data</h2>
<p>In the following examples, we will illustrate how to run <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> using either the easy to use <code>assign.wrapper</code> function for simple analysis or each individual <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> step for more detailed intermediate results.</p>
<p>For either analysis, we will first load <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> and create a temporary directory <code>'tempdir'</code> under the user’s current working directory. All output generated in this vignette will be saved in <code>'tempdir'</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ASSIGN)

<span class="kw">dir.create</span>(<span class="st">"tempdir"</span>)</code></pre></div>
<pre><code>## Warning in dir.create("tempdir"): 'tempdir' already exists</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tempdir &lt;-<span class="st"> "tempdir"</span></code></pre></div>
<p>Next, load the training data, test data, training labels, and test labels. The training dataset is a G (number of genomic measurements) x N (number of samples in pathway perturbation experiments) matrix, including five oncogenic pathways: B-Catenin, E2F3, MYC, RAS, and SRC pathways in this example. The training data labels denote the column indices of control and experimental samples for each perturbation experiment. For example, we specify the column indices of the 10 RAS control samples to be 1:10, and column indices of 10 RAS activated samples to be 39:48. The test dataset is a G (number of genomic measurements) x N (number of patient samples) matrix. The test data labels denote the classes of the N test samples. In our example, test samples 1-53 are adenocarcinoma and samples 54-111 are squamous cell carcinoma. We specify <code>'Adeno'</code> and <code>'Squamous'</code> in the vector of test data labels. Note that the test data labels are optional. <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> outputs additional validation plots to evaluate classification accuracy when test data labels are provided.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(trainingData1)
<span class="kw">data</span>(testData1)
<span class="kw">data</span>(geneList1)
trainingLabel1 &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">control =</span> <span class="kw">list</span>(<span class="dt">bcat=</span><span class="dv">1</span>:<span class="dv">10</span>, <span class="dt">e2f3=</span><span class="dv">1</span>:<span class="dv">10</span>,
                                      <span class="dt">myc=</span><span class="dv">1</span>:<span class="dv">10</span>, <span class="dt">ras=</span><span class="dv">1</span>:<span class="dv">10</span>, <span class="dt">src=</span><span class="dv">1</span>:<span class="dv">10</span>),
                       <span class="dt">bcat =</span> <span class="dv">11</span>:<span class="dv">19</span>, <span class="dt">e2f3 =</span> <span class="dv">20</span>:<span class="dv">28</span>, <span class="dt">myc=</span> <span class="dv">29</span>:<span class="dv">38</span>,
                       <span class="dt">ras =</span> <span class="dv">39</span>:<span class="dv">48</span>, <span class="dt">src =</span> <span class="dv">49</span>:<span class="dv">55</span>)
testLabel1 &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">"Adeno"</span>, <span class="st">"Squamous"</span>), <span class="kw">c</span>(<span class="dv">53</span>,<span class="dv">58</span>))</code></pre></div>
</div>
<div id="run-assign-all-in-one-using-assign-wrapper" class="section level2">
<h2 class="hasAnchor">
<a href="#run-assign-all-in-one-using-assign-wrapper" class="anchor"></a>Run <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> all-in-one using <code>assign.wrapper</code>
</h2>
<p>We developed an all-in-one <code>assign.wrapper</code> function to run <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> with one command. For most users, <code>assign.wrapper</code> will be sufficient. The <code>assign.wrapper</code> function outputs the following files:</p>
<ul>
<li>
<strong>pathway_activity_testset.csv</strong>: <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> predicted pathway activity in test samples.</li>
<li>
<strong>signature_heatmap_testset_prior.pdf</strong>: heatmaps of the expression level of prior signature genes in training samples.</li>
<li>
<strong>pathway_activity_scatterplot_testset.pdf</strong>: scatterplot of pathway activity in test samples. The x-axis represents test samples ordered by pathway activity; the y-axis represents pathway activity.</li>
<li>
<strong>output.rda</strong>: The intermediate results of individual <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> functions.</li>
<li>
<strong>parameters.txt</strong>: A log file containing the parameters used for this <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> run.</li>
</ul>
<p>If training data is provided, <code>assign.wrapper</code> also outputs the following files:</p>
<ul>
<li>
<strong>pathway_activity_trainingset.csv</strong>: <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> predicted pathway activity in training samples.</li>
<li>
<strong>signature_heatmap_trainingset.pdf</strong>: heatmaps of the expression level of signature genes in training samples.</li>
<li>
<strong>pathway_activity_scatterplot_trainingset.pdf</strong>: scatterplot of pathway activity in training samples.</li>
<li>
<strong>signature_gene_list_prior.csv</strong>: the gene list and prior coefficients for the pathway signature.</li>
</ul>
<p>When <code>Adaptive_S</code> is TRUE, <code>assign.wrapper</code> also outputs the following files:</p>
<ul>
<li>
<strong>signature_heatmap_testset_posterior.pdf</strong>: heatmaps of the expression level of posterior signature genes in training samples.</li>
<li>
<strong>posterior_delta.csv</strong>: a csv file of the prior and posterior change in expression and probability of inclusion for each gene in each signature.</li>
<li>
<strong>Signature_convergence.pdf</strong>: A plot of the MCMC convergence.</li>
</ul>
<p>Finally, if the <code>testLabel</code> argument is not NULL, <code>assign.wrapper</code> also outputs the following files:</p>
<ul>
<li>
<strong>pathway_activity_boxplot_testset.pdf</strong>: boxplot of pathway activity in every test class.</li>
</ul>
<p>Here we illustrate how to run <code>assign.wrapper</code> function with three examples. To start, create a temporary directory <code>'tempdir'</code> and load training and test datasets. The individual parameters are described in detail in the sections below and the <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> reference manual.</p>
<div id="example-1-training-data-is-available-but-a-gene-list-of-pathway-signature-genes-is-not-available" class="section level3">
<h3 class="hasAnchor">
<a href="#example-1-training-data-is-available-but-a-gene-list-of-pathway-signature-genes-is-not-available" class="anchor"></a>Example 1: Training data is available, but a gene list of pathway signature genes is not available:</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dir.create</span>(<span class="kw">file.path</span>(tempdir,<span class="st">"wrapper_example1"</span>))
<span class="kw"><a href="../reference/assign.wrapper.html">assign.wrapper</a></span>(<span class="dt">trainingData=</span>trainingData1, <span class="dt">testData=</span>testData1,
               <span class="dt">trainingLabel=</span>trainingLabel1, <span class="dt">testLabel=</span>testLabel1,
               <span class="dt">geneList=</span><span class="ot">NULL</span>, <span class="dt">n_sigGene=</span><span class="kw">rep</span>(<span class="dv">200</span>,<span class="dv">5</span>), <span class="dt">adaptive_B=</span><span class="ot">TRUE</span>,
               <span class="dt">adaptive_S=</span><span class="ot">FALSE</span>, <span class="dt">mixture_beta=</span><span class="ot">TRUE</span>,
               <span class="dt">outputDir=</span><span class="kw">file.path</span>(tempdir,<span class="st">"wrapper_example1"</span>),
               <span class="dt">iter=</span><span class="dv">2000</span>, <span class="dt">burn_in=</span><span class="dv">1000</span>)</code></pre></div>
</div>
<div id="example-2-training-data-is-available-and-a-gene-list-of-pathway-signature-genes-is-available" class="section level3">
<h3 class="hasAnchor">
<a href="#example-2-training-data-is-available-and-a-gene-list-of-pathway-signature-genes-is-available" class="anchor"></a>Example 2: Training data is available, and a gene list of pathway signature genes is available:</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dir.create</span>(<span class="kw">file.path</span>(tempdir,<span class="st">"wrapper_example2"</span>))
<span class="kw"><a href="../reference/assign.wrapper.html">assign.wrapper</a></span>(<span class="dt">trainingData=</span>trainingData1, <span class="dt">testData=</span>testData1,
               <span class="dt">trainingLabel=</span>trainingLabel1, <span class="dt">testLabel=</span><span class="ot">NULL</span>,
               <span class="dt">geneList=</span>geneList1, <span class="dt">n_sigGene=</span><span class="ot">NULL</span>, <span class="dt">adaptive_B=</span><span class="ot">TRUE</span>,
               <span class="dt">adaptive_S=</span><span class="ot">FALSE</span>, <span class="dt">mixture_beta=</span><span class="ot">TRUE</span>,
               <span class="dt">outputDir=</span><span class="kw">file.path</span>(tempdir,<span class="st">"wrapper_example2"</span>),
               <span class="dt">iter=</span><span class="dv">2000</span>, <span class="dt">burn_in=</span><span class="dv">1000</span>)</code></pre></div>
</div>
<div id="example-3-training-data-is-not-available-but-a-gene-list-of-pathway-signature-genes-is-available" class="section level3">
<h3 class="hasAnchor">
<a href="#example-3-training-data-is-not-available-but-a-gene-list-of-pathway-signature-genes-is-available" class="anchor"></a>Example 3: Training data is not available, but a gene list of pathway signature genes is available:</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dir.create</span>(<span class="kw">file.path</span>(tempdir,<span class="st">"wrapper_example3"</span>))
<span class="kw"><a href="../reference/assign.wrapper.html">assign.wrapper</a></span>(<span class="dt">trainingData=</span><span class="ot">NULL</span>, <span class="dt">testData=</span>testData1,
               <span class="dt">trainingLabel=</span><span class="ot">NULL</span>, <span class="dt">testLabel=</span><span class="ot">NULL</span>,
               <span class="dt">geneList=</span>geneList1, <span class="dt">n_sigGene=</span><span class="ot">NULL</span>, <span class="dt">adaptive_B=</span><span class="ot">TRUE</span>,
               <span class="dt">adaptive_S=</span><span class="ot">TRUE</span>, <span class="dt">mixture_beta=</span><span class="ot">TRUE</span>,
               <span class="dt">outputDir=</span><span class="kw">file.path</span>(tempdir,<span class="st">"wrapper_example3"</span>),
               <span class="dt">iter=</span><span class="dv">2000</span>, <span class="dt">burn_in=</span><span class="dv">1000</span>)</code></pre></div>
</div>
</div>
<div id="run-assign-step-by-step" class="section level2">
<h2 class="hasAnchor">
<a href="#run-assign-step-by-step" class="anchor"></a>Run <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> step-by-step</h2>
<p>We developed a series of functions: <code>assign.preprocess</code>, <code>assign.mcmc</code>, <code>assign.convergence</code>, <code>assign.summary</code>, <code>assign.cv.output</code>, and <code>assign.output</code> that work in concert to produce detailed results.</p>
<div id="assign-preprocess" class="section level3">
<h3 class="hasAnchor">
<a href="#assign-preprocess" class="anchor"></a><code>assign.preprocess</code>
</h3>
<p>We first run the <code>assign.preprocess</code> function on the input datasets. When the genomic measurements (e.g., gene expression profiles) of training samples are provided, but predetermined pathway signature gene lists are not provided, the <code>assign.preprocess</code> function utilizes a Bayesian univariate regression module to select a gene set (usually 50-200 genes, but this can be specified by the user) based on the absolute value of the regression coefficient (fold change) and the posterior probability of the variable to be selected (statistical significance). Since we have no predetermined gene lists to provide, we leave the <code>geneList</code> option as default NULL. Here we specify 200 signature genes for each of the five pathways.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># training dataset is available;</span>
<span class="co"># the gene list of pathway signature is NOT available</span>
processed.data &lt;-<span class="st"> </span><span class="kw"><a href="../reference/assign.preprocess.html">assign.preprocess</a></span>(<span class="dt">trainingData=</span>trainingData1,
                                    <span class="dt">testData=</span>testData1,
                                    <span class="dt">trainingLabel=</span>trainingLabel1,
                                    <span class="dt">geneList=</span><span class="ot">NULL</span>, <span class="dt">n_sigGene=</span><span class="kw">rep</span>(<span class="dv">200</span>,<span class="dv">5</span>))</code></pre></div>
<p>Alternatively, the users can have both the training data and the curated/predetermined pathway signatures. Some genes in the curated pathway signatures, although not significantly differentially expressed, need to be included for the purpose of prediction. In this case, we specify the <code>trainingData</code> and <code>geneList</code> parameters when both the training dataset and predetermined signature gene list are available.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># training dataset is available;</span>
<span class="co"># the gene list of pathway signature is available</span>
processed.data &lt;-<span class="st"> </span><span class="kw"><a href="../reference/assign.preprocess.html">assign.preprocess</a></span>(<span class="dt">trainingData=</span>trainingData1,
                                    <span class="dt">testData=</span>testData1,
                                    <span class="dt">trainingLabel=</span>trainingLabel1,
                                    <span class="dt">geneList=</span>geneList1)</code></pre></div>
<p>In some cases, the expression profiles (training dataset) is unavailable. Only the knowledge-based gene list or gene list from the joint knowledge of some prior profiling experiments is available. In this case, we specify <code>geneList</code> and leave the <code>trainingData</code> and <code>trainingLabel</code> as default NULL.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># training dataset is NOT available;</span>
<span class="co"># the gene list of pathway signature is available</span>
processed.data &lt;-<span class="st"> </span><span class="kw"><a href="../reference/assign.preprocess.html">assign.preprocess</a></span>(<span class="dt">trainingData=</span><span class="ot">NULL</span>,
                                    <span class="dt">testData=</span>testData1,
                                    <span class="dt">trainingLabel=</span><span class="ot">NULL</span>,
                                    <span class="dt">geneList=</span>geneList1)</code></pre></div>
<p>The <code>assign.preprocess</code> function returns the processed training dataset (<code>trainingData_sub</code>) and test dataset (<code>testData_sub</code>) as well as the prior parameters for the background vector (<code>B_vector</code>), signature matrix (<code>S_matrix</code>) and the probability signature matrix (<code>Pi_matrix</code>) and differentially expressed gene lists of each pathway (<code>diffGeneList</code>). The details of the <code>assign.preprocess</code> output are described in the <code>'value'</code> section of the manual page of <code>assign.preprocess</code> function. The output data of <code>assign.preprocess</code> function are used as the input data of the <code>assign.mcmc</code> function.</p>
</div>
<div id="assign-mcmc" class="section level3">
<h3 class="hasAnchor">
<a href="#assign-mcmc" class="anchor"></a><code>assign.mcmc</code>
</h3>
<p>For the <code>assign.mcmc</code> function, Y, Bg, and X are specified as the output of the <code>assign.preprocess</code> function. The <code>adaptive_B</code> (adaptive background), <code>Adaptive_S</code> (adaptive signature) and <code>mixture_beta</code> (regularization of signature strength) can be specified TRUE or FALSE based on the analysis context. When training and test samples are from the different cell or tissue types, we recommend the adaptive background option to be TRUE. Notice that when the training dataset is not available, the adaptive signature option must be set TRUE, meaning that the magnitude of the signature should be estimated from the test dataset. The default <code>iter</code> (iteration) is 2000. Particularly, when training datasets are unavailable, it is better to specify the <code>X</code> option in the <code>assign.mcmc</code> using a more informative X (specify up- or down- regulated genes) to initiate the model.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mcmc.chain &lt;-<span class="st"> </span><span class="kw"><a href="../reference/assign.mcmc.html">assign.mcmc</a></span>(<span class="dt">Y=</span>processed.data$testData_sub,
                          <span class="dt">Bg =</span> processed.data$B_vector,
                          <span class="dt">X=</span>processed.data$S_matrix,
                          <span class="dt">Delta_prior_p =</span> processed.data$Pi_matrix,
                          <span class="dt">iter =</span> <span class="dv">2000</span>, <span class="dt">adaptive_B=</span><span class="ot">TRUE</span>,
                          <span class="dt">adaptive_S=</span><span class="ot">FALSE</span>, <span class="dt">mixture_beta=</span><span class="ot">TRUE</span>)</code></pre></div>
<p>The <code>assign.mcmc</code> function returns the MCMC chain recording default 2000 iterations for each parameter. The details of <code>assign.mcmc</code> output are described in the <code>'value'</code> section of the manual page of <code>assign.mcmc</code> function.</p>
</div>
<div id="assign-convergence" class="section level3">
<h3 class="hasAnchor">
<a href="#assign-convergence" class="anchor"></a><code>assign.convergence</code>
</h3>
<p>We can make a trace plot to check the convergence of the model parameters through <code>assign.convergence</code>. The <code>burn_in</code> default is 0, so that the trace plot starts from the first iteration. Additional iterations can be specified if the MCMC chain does not converge in 2000 iterations.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">trace.plot &lt;-<span class="st"> </span><span class="kw"><a href="../reference/assign.convergence.html">assign.convergence</a></span>(<span class="dt">test=</span>mcmc.chain, <span class="dt">burn_in=</span><span class="dv">0</span>, <span class="dt">iter=</span><span class="dv">2000</span>,
                                 <span class="dt">parameter=</span><span class="st">"B"</span>, <span class="dt">whichGene=</span><span class="dv">1</span>,
                                 <span class="dt">whichSample=</span><span class="ot">NA</span>, <span class="dt">whichPath=</span><span class="ot">NA</span>)</code></pre></div>
<p>The <code>assign.convergence</code> function returns a vector of the estimated values from each Gibbs sampling iteration of the model parameter to be checked and a trace plot of the parameter.</p>
</div>
<div id="assign-summary" class="section level3">
<h3 class="hasAnchor">
<a href="#assign-summary" class="anchor"></a><code>assign.summary</code>
</h3>
<p>We then apply the <code>assign.summary</code> function to compute the posterior mean of each parameter. Typically we use the second half of the MCMC chain to compute the posterior mean. We specify the default burn-in period to be the first 1000 iteration and the default total iteration to be 2000. The 1000 burn-in iterations are discarded when we compute the posterior mean. The <code>adaptive_B</code>, <code>Adaptive_S</code> and <code>mixture_beta</code> options have to set the same as those in the <code>assign.mcmc</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mcmc.pos.mean &lt;-<span class="st"> </span><span class="kw"><a href="../reference/assign.summary.html">assign.summary</a></span>(<span class="dt">test=</span>mcmc.chain, <span class="dt">burn_in=</span><span class="dv">1000</span>,
                                <span class="dt">iter=</span><span class="dv">2000</span>, <span class="dt">adaptive_B=</span><span class="ot">TRUE</span>,
                                <span class="dt">adaptive_S=</span><span class="ot">FALSE</span>, <span class="dt">mixture_beta=</span><span class="ot">TRUE</span>)</code></pre></div>
<p>The <code>assign.summary</code> function returns the posterior mean of each parameter. The details of the <code>assign.summary</code> output are described in the <code>'value'</code> section of the manual page of <code>assign.summary</code> function.</p>
</div>
<div id="assign-cv-output" class="section level3">
<h3 class="hasAnchor">
<a href="#assign-cv-output" class="anchor"></a><code>assign.cv.output</code>
</h3>
<p>The <code>assign.cv.output</code> generates the cross-validation results in the training samples. Output files from <code>assign.cv.output</code> are:</p>
<ul>
<li>
<strong>pathway_activity_trainingset.csv</strong>: <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> predicted pathway activity in training samples.</li>
<li>
<strong>signature_heatmap_trainingset.pdf</strong>: heatmaps of the expression level of signature genes in training samples.</li>
<li>
<strong>pathway_activity_scatterplot_trainingset.pdf</strong>: scatterplot of pathway activity in training samples.</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># For cross-validation, Y in the assign.mcmc function</span>
<span class="co"># should be specified as processed.data$trainingData_sub.</span>
<span class="kw"><a href="../reference/assign.cv.output.html">assign.cv.output</a></span>(<span class="dt">processed.data=</span>processed.data,
                 <span class="dt">mcmc.pos.mean.trainingData=</span>mcmc.pos.mean,
                 <span class="dt">trainingData=</span>trainingData1,
                 <span class="dt">trainingLabel=</span>trainingLabel1, <span class="dt">adaptive_B=</span><span class="ot">FALSE</span>,
                 <span class="dt">adaptive_S=</span><span class="ot">FALSE</span>, <span class="dt">mixture_beta=</span><span class="ot">TRUE</span>,
                 <span class="dt">outputDir=</span>tempdir)</code></pre></div>
</div>
<div id="assign-output" class="section level3">
<h3 class="hasAnchor">
<a href="#assign-output" class="anchor"></a><code>assign.output</code>
</h3>
<p>The <code>assign.output</code> generates the prediction results in the test samples. Output files from <code>assign.output</code> are:</p>
<ul>
<li>
<strong>pathway_activity_testset.csv</strong>: <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> predicted pathway activity in test samples.</li>
<li>
<strong>signature_heatmap_testset_prior.pdf</strong>: heatmaps of the expression level of prior signature genes in training samples.</li>
<li>
<strong>signature_heatmap_testset_posterior.pdf</strong>: heatmaps of the expression level of posterior signature genes in training samples. This plot is only generated when <code>Adaptive_S</code> is specified TRUE.</li>
<li>
<strong>pathway_activity_scatterplot_testset.pdf</strong>: scatterplot of pathway activity in test samples. The x-axis represents test samples ordered by pathway activity; the y-axis represents pathway activity.</li>
<li>
<strong>pathway_activity_boxplot_testset.pdf</strong>: boxplot of pathway activity in every test class. This plot is only generated only the <code>testLabel</code> argument is not NULL.</li>
</ul>
<p>The user needs to specify the output directory in the <code>outputDir</code> option, when running <code>assign.cv.output</code> and <code>assign.output</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/assign.output.html">assign.output</a></span>(<span class="dt">processed.data=</span>processed.data,
              <span class="dt">mcmc.pos.mean.testData=</span>mcmc.pos.mean,
              <span class="dt">trainingData=</span>trainingData1, <span class="dt">testData=</span>testData1,
              <span class="dt">trainingLabel=</span>trainingLabel1,
              <span class="dt">testLabel=</span>testLabel1, <span class="dt">geneList=</span><span class="ot">NULL</span>,
              <span class="dt">adaptive_B=</span><span class="ot">TRUE</span>, <span class="dt">adaptive_S=</span><span class="ot">FALSE</span>,
              <span class="dt">mixture_beta=</span><span class="ot">TRUE</span>, <span class="dt">outputDir=</span>tempdir)</code></pre></div>
</div>
</div>
</div>
<div id="additional-features" class="section level1">
<h1 class="hasAnchor">
<a href="#additional-features" class="anchor"></a>Additional Features</h1>
<div id="anchor-gene-lists-and-exclude-gene-lists" class="section level2">
<h2 class="hasAnchor">
<a href="#anchor-gene-lists-and-exclude-gene-lists" class="anchor"></a>Anchor Gene Lists and Exclude Gene Lists</h2>
<p>The <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> package allows a signature to be adapted to fit other biological contexts, reducing the contribution of specific genes in the signature to better match heterogeneity observed in the test dataset. Occasionally, adapting a signature may reduce the importance of key signature genes. For example, if a signature is created by overexpressing an oncogenic gene in a cell line, but during the adaptation step, <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> reduces the importance of that key gene, the quality of the <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> predictions may be reduced. Alternatively, if a gene in the signature is associated with some other heterogeneity in the data, such as smoking status, <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> may adapt to differences in that gene, rather than the actual desired signature activity predictions. To this end, we have added the ability to provide a list of key genes to anchor in the signature, and genes to exclude from the signature. <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> accomplishes this by setting the probability of inclusion into the signature to one for anchor genes, and zero for exclude genes. The change in expression values can still adapt, increasing or reducing the fold change associated with each gene in the signature, but the anchor genes will always contribute to the final signature, and the exclude genes will not.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dir.create</span>(<span class="kw">file.path</span>(tempdir, <span class="st">"anchor_exclude_example"</span>))

anchorList =<span class="st"> </span><span class="kw">list</span>(<span class="dt">bcat=</span><span class="st">"224321_at"</span>,
                  <span class="dt">e2f3=</span><span class="st">"202589_at"</span>,
                  <span class="dt">myc=</span><span class="st">"221891_x_at"</span>,
                  <span class="dt">ras=</span><span class="st">"201820_at"</span>,
                  <span class="dt">src=</span><span class="st">"224567_x_at"</span>)
excludeList =<span class="st"> </span><span class="kw">list</span>(<span class="dt">bcat=</span><span class="st">"1555340_x_at"</span>,
                   <span class="dt">e2f3=</span><span class="st">"1555340_x_at"</span>,
                   <span class="dt">myc=</span><span class="st">"1555340_x_at"</span>,
                   <span class="dt">ras=</span><span class="st">"204748_at"</span>,
                   <span class="dt">src=</span><span class="st">"1555339_at"</span>)

<span class="kw"><a href="../reference/assign.wrapper.html">assign.wrapper</a></span>(<span class="dt">trainingData=</span>trainingData1, <span class="dt">testData=</span>testData1,
               <span class="dt">trainingLabel=</span>trainingLabel1, <span class="dt">testLabel=</span><span class="ot">NULL</span>,
               <span class="dt">geneList=</span>geneList1, <span class="dt">n_sigGene=</span><span class="ot">NULL</span>, <span class="dt">adaptive_B=</span><span class="ot">TRUE</span>,
               <span class="dt">adaptive_S=</span><span class="ot">TRUE</span>, <span class="dt">mixture_beta=</span><span class="ot">TRUE</span>,
               <span class="dt">outputDir=</span><span class="kw">file.path</span>(tempdir, <span class="st">"anchor_exclude_example"</span>),
               <span class="dt">anchorGenes=</span>anchorList, <span class="dt">excludeGenes=</span>excludeList,
               <span class="dt">iter=</span><span class="dv">2000</span>, <span class="dt">burn_in=</span><span class="dv">1000</span>)</code></pre></div>
</div>
<div id="fraction-of-upregulated-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#fraction-of-upregulated-genes" class="anchor"></a>Fraction of Upregulated Genes</h2>
<p>By default, <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> Bayesian gene selection chooses the signature genes with an equal fraction of genes that increase with pathway activity and genes that decrease with pathway activity. Use the pctUp parameter to modify this fraction. Set pctUP to NULL to select the most significant genes, regardless of direction.</p>
</div>
</div>
<div id="gfrn-optimization-procedure" class="section level1">
<h1 class="hasAnchor">
<a href="#gfrn-optimization-procedure" class="anchor"></a>GFRN Optimization Procedure</h1>
<p>When running <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em>, the number of genes in the gene list can affect the predictions that <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> produces, but it is not always clear how long the gene list should be. Included within <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> is the optimization procedure used in the publication <a href="http://dx.doi.org/10.1186/s13073-017-0429-x">Activity of distinct growth factor receptor network components in breast tumors uncovers two biologically relevant subtypes</a>. The function allows you to optimize the gene list lengths for the pathways included in the paper using your own correlation data and gene list lengths. This function runs <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> pathway prediction on various gene list lengths to find the optimum gene list length for the GFRN pathways by correlating the <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> predictions to a matrix of correlation data that you provide. This function takes a long time to run because you are running <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> many times on many pathways, so I recommend parallelizing by pathway or running the <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> predictions first (long and parallelizable) and then running the correlation step (quick) separately.</p>
<div id="example-optimization-procedure" class="section level2">
<h2 class="hasAnchor">
<a href="#example-optimization-procedure" class="anchor"></a>Example Optimization Procedure</h2>
<p>The following example optimizes the pathway length for the AKT pathway based on correlating <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> predictions with proteomics data. First, read in the test data that you want to predict using <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> and the data (e.g. proteomics data) that will be used for correlation:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dir.create</span>(<span class="kw">file.path</span>(tempdir, <span class="st">"optimization_example"</span>))
<span class="kw">setwd</span>(<span class="kw">file.path</span>(tempdir, <span class="st">"optimization_example"</span>))

testData &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">"https://www.dropbox.com/s/6iuxcr2f0fjbirv/icbp_Rsubread_tpmlog.txt?dl=1"</span>,
                       <span class="dt">sep=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, <span class="dt">row.names=</span><span class="dv">1</span>, <span class="dt">header=</span><span class="dv">1</span>)

corData &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">"https://www.dropbox.com/s/nm3m38ca6v4lu10/proteomics.txt?dl=1"</span>,
                      <span class="dt">sep=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, <span class="dt">row.names=</span><span class="dv">1</span>, <span class="dt">header=</span><span class="dv">1</span>)</code></pre></div>
<p>Next, create a list of data used for correlation. The list should contain a vector of column names from the correlation data for each of the pathways that are being optimized. The gene list length that has the largest average correlation for the columns in the correlation list will be the optimized gene list.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#this is a list of pathways and columns in the correlation data that will</span>
<span class="co">#be used for correlation</span>
corList &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">akt=</span><span class="kw">c</span>(<span class="st">"Akt"</span>,<span class="st">"PDK1"</span>,<span class="st">"PDK1p241"</span>))</code></pre></div>
<p>Finally, run the ComBat batch correction procedure and run the <code>optimizeGFRN</code> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#run the batch correction procedure between the test and training data</span>
combat.data &lt;-<span class="st"> </span><span class="kw"><a href="../reference/ComBat.step2.html">ComBat.step2</a></span>(testData, <span class="dt">pcaPlots =</span> <span class="ot">TRUE</span>)

<span class="co">#run the default optimization procedure</span>
optimization_results &lt;-<span class="st"> </span><span class="kw"><a href="../reference/optimizeGFRN.html">optimizeGFRN</a></span>(combat.data, corData,
                                     corList, <span class="dt">run=</span><span class="st">"akt"</span>)</code></pre></div>
<p><em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> will output the results for each gene list length in the current working directory. The <code>optimizeGFRN</code> function returns a list of optimized gene lists which can be used on other datasets and correlation results. Additional options and documentation is available in the <code>optimizeGFRN</code> function documentation.</p>
</div>
</div>
<div id="citing-assign" class="section level1">
<h1 class="hasAnchor">
<a href="#citing-assign" class="anchor"></a>Citing <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em>
</h1>
<p>If you use <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> in your publication, please cite:</p>
<p>Shen, Y. et al. ASSIGN: context-specific genomic profiling of multiple heterogeneous biological pathways. Bioinformatics 31 (11), 1745-1753 (2015). <a href="http://dx.doi.org/10.1093/bioinformatics/btv031">doi:10.1093/bioinformatics/btv031</a></p>
</div>
<div id="conclusion" class="section level1">
<h1 class="hasAnchor">
<a href="#conclusion" class="anchor"></a>Conclusion</h1>
<p>Please see the <em><a href="http://bioconductor.org/packages/ASSIGN">ASSIGN</a></em> reference manual for full descriptions of functions and the various options they support.</p>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<pre><code>## R version 3.4.2 (2017-09-28)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS High Sierra 10.13.2
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] ASSIGN_1.15.1
## 
## loaded via a namespace (and not attached):
##  [1] compiler_3.4.2  backports_1.1.1 BiocStyle_2.6.1 magrittr_1.5   
##  [5] rprojroot_1.2   tools_3.4.2     htmltools_0.3.6 yaml_2.1.15    
##  [9] Rcpp_0.12.14    stringi_1.1.6   rmarkdown_1.8   knitr_1.17     
## [13] stringr_1.2.0   digest_0.6.12   evaluate_0.10.1</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li>
<a href="#how-to-use-the-assign-package">How to use the ASSIGN package</a><ul class="nav nav-pills nav-stacked">
<li><a href="#example-data">Example Data</a></li>
      <li><a href="#run-assign-all-in-one-using-assign-wrapper">Run <em><a href="http://bioconductor-org/packages/ASSIGN">ASSIGN</a></em> all-in-one using <code>assign.wrapper</code></a></li>
      <li><a href="#run-assign-step-by-step">Run <em><a href="http://bioconductor-org/packages/ASSIGN">ASSIGN</a></em> step-by-step</a></li>
      </ul>
</li>
      <li>
<a href="#additional-features">Additional Features</a><ul class="nav nav-pills nav-stacked">
<li><a href="#anchor-gene-lists-and-exclude-gene-lists">Anchor Gene Lists and Exclude Gene Lists</a></li>
      <li><a href="#fraction-of-upregulated-genes">Fraction of Upregulated Genes</a></li>
      </ul>
</li>
      <li>
<a href="#gfrn-optimization-procedure">GFRN Optimization Procedure</a><ul class="nav nav-pills nav-stacked">
<li><a href="#example-optimization-procedure">Example Optimization Procedure</a></li>
      </ul>
</li>
      <li><a href="#citing-assign">Citing <em><a href="http://bioconductor-org/packages/ASSIGN">ASSIGN</a></em></a></li>
      <li><a href="#conclusion">Conclusion</a></li>
      <li><a href="#session-info">Session info</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Ying Shen, Andrea H. Bild, W. Evan Johnson, Mumtehena Rahman, David Jenkins.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
