<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>The Gibbs sampling algorithm to approximate the joint distribution of the
model parameters — assign.mcmc • ASSIGN</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>
  
  
<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">ASSIGN</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/ASSIGN.vignette.html">Primer</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/compbiomed/ASSIGN">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>The Gibbs sampling algorithm to approximate the joint distribution of the
model parameters</h1>
    </div>

    
    <p>The assign.mcmc function uses a Bayesian sparse factor analysis model to
estimate the adaptive baseline/background, adaptive pathway signature, and
pathway activation status of individual test (disease) samples.</p>
    

    <pre class="usage"><span class='fu'>assign.mcmc</span>(<span class='no'>Y</span>, <span class='no'>Bg</span>, <span class='no'>X</span>, <span class='no'>Delta_prior_p</span>, <span class='kw'>iter</span> <span class='kw'>=</span> <span class='fl'>2000</span>, <span class='kw'>adaptive_B</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>adaptive_S</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>mixture_beta</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>sigma_sZero</span> <span class='kw'>=</span> <span class='fl'>0.01</span>,
  <span class='kw'>sigma_sNonZero</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>p_beta</span> <span class='kw'>=</span> <span class='fl'>0.01</span>, <span class='kw'>sigma_bZero</span> <span class='kw'>=</span> <span class='fl'>0.01</span>,
  <span class='kw'>sigma_bNonZero</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>alpha_tau</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>beta_tau</span> <span class='kw'>=</span> <span class='fl'>0.01</span>,
  <span class='kw'>Bg_zeroPrior</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>S_zeroPrior</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>ECM</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a> Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>Y</th>
      <td><p>The G x J matrix of genomic measures (i.g., gene expression) of test
samples. Y is the testData_sub variable returned from the data.process
function. Genes/probes present in at least one pathway signature are
retained.</p></td>
    </tr>
    <tr>
      <th>Bg</th>
      <td><p>The G x 1 vector of genomic measures of the baseline/background
(B). Bg is the B_vector variable returned from the data.process function. Bg
is the starting value of baseline/background level in the MCMC chain.</p></td>
    </tr>
    <tr>
      <th>X</th>
      <td><p>The G x K matrix of genomic measures of the signature. X is the
S_matrix variable returned from the data.process function. X is the starting
value of pathway signatures in the MCMC chain.</p></td>
    </tr>
    <tr>
      <th>Delta_prior_p</th>
      <td><p>The G x K matrix of prior probability of a gene being
"significant" in its associated pathway. Delta_prior_p is the Pi_matrix
variable returned from the data.process function.</p></td>
    </tr>
    <tr>
      <th>iter</th>
      <td><p>The number of iterations in the MCMC. The default is 2000.</p></td>
    </tr>
    <tr>
      <th>adaptive_B</th>
      <td><p>Logicals. If TRUE, the model adapts the
baseline/background (B) of genomic measures for the test samples. The
default is TRUE.</p></td>
    </tr>
    <tr>
      <th>adaptive_S</th>
      <td><p>Logicals. If TRUE, the model adapts the signatures (S) of
genomic measures for the test samples. The default is FALSE.</p></td>
    </tr>
    <tr>
      <th>mixture_beta</th>
      <td><p>Logicals. If TRUE, elements of the pathway activation
matrix are modeled by a spike-and-slab mixture distribution. The default is
TRUE.</p></td>
    </tr>
    <tr>
      <th>sigma_sZero</th>
      <td><p>Each element of the signature matrix (S) is modeled by a
spike-and-slab mixture distribution. Sigma_sZero is the variance of the
spike normal distribution. The default is 0.01.</p></td>
    </tr>
    <tr>
      <th>sigma_sNonZero</th>
      <td><p>Each element of the signature matrix (S) is modeled by
a spike-and-slab mixture distribution. Sigma_sNonZero is the variance of
the slab normal distribution. The default is 1.</p></td>
    </tr>
    <tr>
      <th>p_beta</th>
      <td><p>p_beta is the prior probability of a pathway being activated
in individual test samples. The default is 0.01.</p></td>
    </tr>
    <tr>
      <th>sigma_bZero</th>
      <td><p>Each element of the pathway activation matrix (A) is
modeled by a spike-and-slab mixture distribution. sigma_bZero is the
variance of the spike normal distribution. The default is 0.01.</p></td>
    </tr>
    <tr>
      <th>sigma_bNonZero</th>
      <td><p>Each element of the pathway activation matrix (A) is
modeled by a spike-and-slab mixture distribution. sigma_bNonZero is the
variance of the slab normal distribution. The default is 1.</p></td>
    </tr>
    <tr>
      <th>alpha_tau</th>
      <td><p>The shape parameter of the precision (inverse of the
variance) of a gene. The default is 1.</p></td>
    </tr>
    <tr>
      <th>beta_tau</th>
      <td><p>The rate parameter of the precision (inverse of the
variance) of a gene. The default is 0.01.</p></td>
    </tr>
    <tr>
      <th>Bg_zeroPrior</th>
      <td><p>Logicals. If TRUE, the prior distribution of
baseline/background level follows a normal distribution with mean zero. The
default is TRUE.</p></td>
    </tr>
    <tr>
      <th>S_zeroPrior</th>
      <td><p>Logicals. If TRUE, the prior distribution of signature
follows a normal distribution with mean zero. The default is TRUE.</p></td>
    </tr>
    <tr>
      <th>ECM</th>
      <td><p>Logicals. If TRUE, ECM algorithm, rather than Gibbs sampling, is
applied to approximate the model parameters. The default is FALSE.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    

<dt>beta_mcmc</dt><dd><p>The iter x K x J array of the pathway activation
level estimated in every iteration of MCMC.</p></dd> <dt>tau2_mcmc</dt><dd><p>The iter x G
matrix of the precision of genes estimated in every iteration of MCMC</p></dd>
<dt>gamma_mcmc</dt><dd><p>The iter x K x J array of probability of pathway being
activated estimated in every iteration of MCMC.</p></dd> <dt>kappa_mcmc</dt><dd><p>The iter
x K x J array of pathway activation level (adjusted beta scaling between 0
and 1) estimated in every iteration of MCMC.)</p></dd> <dt>S_mcmc</dt><dd><p>The iter x G x
K array of signature estimated in every iteration of MCMC.</p></dd>
<dt>Delta_mcmc</dt><dd><p>The iter x G x K array of binary indicator of a gene being
significant estimated in every iteration of MCMC.</p></dd>

    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The assign.mcmc function can be set as following major modes. The
combination of logical values of adaptive_B, adaptive_S and mixture_beta can
form different modes.</p>
<p>Mode A: adaptive_B = FALSE, adaptive_S = FALSE, mixture_beta = FALSE. This
is a regression mode without adaptation of baseline/background, signature,
and no shrinkage of the pathway activation level.</p>
<p>Mode B: adaptive_B = TRUE, adaptive_S = FALSE, mixture_beta = FALSE. This is
a regression mode with adaptation of baseline/background, but without
signature, and with no shrinkage of the pathway activation level.</p>
<p>Mode C: adaptive_B = TRUE, adaptive_S = FALSE, mixture_beta = TRUE. This is
a regression mode with adaptation of baseline/background, but without
signature, and with shrinkage of the pathway activation level when it is not
significantly activated.</p>
<p>Mode D: adaptive_B = TRUE, adaptive_S = TRUE, mixture_beta = TRUE. This is a
Bayesian factor analysis mode with adaptation of baseline/background,
adaptation signature, and with shrinkage of the pathway activation level.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
</div><div class='input'><span class='no'>mcmc.chain</span> <span class='kw'>&lt;-</span> <span class='fu'>assign.mcmc</span>(<span class='kw'>Y</span><span class='kw'>=</span><span class='no'>processed.data</span>$<span class='no'>testData_sub</span>,
                          <span class='kw'>Bg</span> <span class='kw'>=</span> <span class='no'>processed.data</span>$<span class='no'>B_vector</span>,
                          <span class='kw'>X</span><span class='kw'>=</span><span class='no'>processed.data</span>$<span class='no'>S_matrix</span>,
                          <span class='kw'>Delta_prior_p</span> <span class='kw'>=</span> <span class='no'>processed.data</span>$<span class='no'>Pi_matrix</span>,
                          <span class='kw'>iter</span> <span class='kw'>=</span> <span class='fl'>20</span>, <span class='kw'>adaptive_B</span><span class='kw'>=</span><span class='fl'>TRUE</span>, <span class='kw'>adaptive_S</span><span class='kw'>=</span><span class='fl'>FALSE</span>,
                          <span class='kw'>mixture_beta</span><span class='kw'>=</span><span class='fl'>TRUE</span>)</div><div class='output co'>#&gt; <span class='message'>Start Gibbs sampling...</span></div><div class='output co'>#&gt; <span class='message'>| 0%                                  50%                                 100% |</span></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    
Ying Shen

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Ying Shen, Andrea H. Bild, W. Evan Johnson, Mumtehena Rahman, David Jenkins.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
