<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Summary of the model parameters estimated by the Gibbs sampling algorithm — assign.summary • ASSIGN</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>
  
  
<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">ASSIGN</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/ASSIGN.vignette.html">Primer</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/compbiomed/ASSIGN">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Summary of the model parameters estimated by the Gibbs sampling algorithm</h1>
    </div>

    
    <p>The assign.summary function computes the posterior mean of the model
parameters estimated in every iteration during the Gibbs sampling.</p>
    

    <pre class="usage"><span class='fu'>assign.summary</span>(<span class='no'>test</span>, <span class='kw'>burn_in</span> <span class='kw'>=</span> <span class='fl'>1000</span>, <span class='kw'>iter</span> <span class='kw'>=</span> <span class='fl'>2000</span>, <span class='kw'>adaptive_B</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>adaptive_S</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>mixture_beta</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a> Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>test</th>
      <td><p>The list object returned from the assign.mcmc function. The list
components are the MCMC chains of the B, S, Delta, beta, gamma, and sigma.</p></td>
    </tr>
    <tr>
      <th>burn_in</th>
      <td><p>The number of burn-in iterations. These iterations are
discarded when computing the posterior means of the model parameters. The
default is 1000.</p></td>
    </tr>
    <tr>
      <th>iter</th>
      <td><p>The number of total iterations. The default is 2000.</p></td>
    </tr>
    <tr>
      <th>adaptive_B</th>
      <td><p>Logicals. If TRUE, the model adapts the
baseline/background (B) of genomic measures for the test samples. The
default is TRUE.</p></td>
    </tr>
    <tr>
      <th>adaptive_S</th>
      <td><p>Logicals. If TRUE, the model adapts the signatures (S) of
genomic measures for the test samples. The default is FALSE.</p></td>
    </tr>
    <tr>
      <th>mixture_beta</th>
      <td><p>Logicals. If TRUE, elements of the pathway activation
matrix are modeled by a spike-and-slab mixuture distribution. The default is
TRUE.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    

<dt>beta_pos</dt><dd><p>The N x K matrix of the posterior mean of the
pathway activation level in test samples (transposed matrix A). Columns:K
pathways; rows: N test samples</p></dd> <dt>sigma_pos</dt><dd><p>The G x 1 vector of the
posterior mean of the variance of gene.</p></dd> <dt>kappa_pos</dt><dd><p>The N x K matrix
of posterior mean of pathway activation level in test samples (transposed
matrix A) (adjusted beta_pos scaling between 0 and 1). Columns:K pathways;
rows: N test samples</p></dd> <dt>gamma_pos</dt><dd><p>The N x K matrix of the posterior
probability of pathways being activated in test samples.</p></dd> <dt>S_pos</dt><dd><p>The G
x K matrix of the posterior mean of pathway signature genes.</p></dd>
<dt>Delta_pos</dt><dd><p>The G x K matrix of the posterior probability of genes
being significant in the associated pathways.</p></dd>

    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The assign.summary function is suggested to run after the assign.convergence
function, which is used to check the convergency of the MCMC chain. If the
MCMC chain does not converge to a stationary phase, more iterations are
required in the assign.mcmc function. The number of burn-in iterations is
usually set to be half of the number of total iterations, meaning that the
first half of the MCMC chain is discarded when computing the posterior
means.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    
Ying Shen

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Ying Shen, Andrea H. Bild, W. Evan Johnson, Mumtehena Rahman, David Jenkins.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
